lib/routes/oauth.js

```diff
@@ -140,8 +145,68 @@ router.get('/redirect', function (req, res) {
 
     log.error('msg', msg);
 
-    res.send(400, msg);
+    return res.send(400, msg);
   }
 });
 
+/**
+ * Request user profile
+ *
+ * @param token
+ * @returns {Promise}
+ */
+function requestProfile(token) {
+  return new Promise(function (resolve, reject) {
+    request.get({
+      uri: fxaOAuthConfig.profile_uri + '/profile',
+      headers: {
+        Authorization: 'Bearer ' + token
+      }
+    }, function (err, r, body) {
+      log.verbose(err, body);
+
+      if (err || r.statusCode >= 400) {
+        return reject({status: 400, err: err || body });
+      }
+
+      var data = JSON.parse(body);
+
+      return resolve({
+        email: data.email,
+        uid: data.uid
+      });
+    });
+
+  });
+}
+
+/**
+ * Activate the developer
+ *
+ * @param token
+ * @returns {Promise}
+ */
+function activateDeveloper(token) {
+  return new Promise(function (resolve, reject) {
+    request.post({
+      uri: fxaOAuthConfig.oauth_internal_uri + '/developer/activate',
+      headers: {
+        Authorization: 'Bearer ' + token
+      }
+    }, function (err, r, body) {
+      log.verbose(err, body);
+
+      if (err || r.statusCode >= 400) {
+        return reject({status: 400, err: err});
+      }
+
+      var developerData = JSON.parse(body);
```

Comment:
IIRC, you can also set `json` to an object and it sets the `body` and some headers.

> `json` - sets `body` but to JSON representation of value and adds `Content-type: application/json` header. Additionally, parses the response body as JSON.
> &mdash; via https://github.com/request/request#requestoptions-callback